tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -12202,10 +12201,10 @@ DiagnoseElementTypes(Sema &S, SourceLocation Loc, QualType Ty, bool &Empty,
     return false;
   case AR_TOBJ_OBJECT:
     Empty = false;
-    if (AllowObjectInContext(Ty, DiagContext))
+    if (!CheckObjects || AllowObjectInContext(Ty, ObjDiagContext))
```

Comment:
I didn't expect and after review didn't observe any cases where the call to this would result in `CheckObjects` being false. There's nothing wrong with handling the condition in the interest of making it more robust, but I wanted to confirm I had that right.